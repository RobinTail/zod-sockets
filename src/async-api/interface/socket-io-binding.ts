import type { ReferenceObject, SchemaObject } from "openapi3-ts/oas31";
import {
  WSChannelBinding,
  WSMessageBinding,
  WSOperationBinding,
  WSServerBinding,
} from "./websocket-binding";

export interface SocketIOServerBinding extends WSServerBinding {}
export interface SocketIOChannelBinding extends WSChannelBinding {}
export interface SocketIOOperationBinding extends WSOperationBinding {}
export interface SocketIOMessageBinding extends WSMessageBinding {
  /** @desc Acknowledgement schema */
  ack?: SchemaObject | ReferenceObject;
}
