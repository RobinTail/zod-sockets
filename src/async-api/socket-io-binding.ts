import type { ReferenceObject, SchemaObject } from "openapi3-ts/oas31";

// @todo use 1.0.0 when production ready
export const version = "0.11.0";

/**
 * @desc Protocol-specific definitions for a server.
 * @desc This object MUST NOT contain any properties. Its name is reserved for future use.
 */
export interface SocketIOServerBinding {}

/** @desc Protocol-specific definitions for a channel. */
export interface SocketIOChannelBinding {
  /**
   * @desc The HTTP method to use when establishing the connection. Its value MUST be either GET or POST.
   * */
  method: "GET" | "POST";
  /**
   * @desc A Schema object containing the definitions for each query parameter.
   * @desc This schema MUST be of type object and have a properties key.
   */
  query: SchemaObject | ReferenceObject;
  /**
   * @desc A Schema object containing the definitions of the HTTP headers to use when establishing the connection.
   * @desc This schema MUST be of type object and have a properties key.
   */
  headers: SchemaObject | ReferenceObject;
  /**
   * @desc The version of this binding. If omitted, "latest" MUST be assumed.
   */
  bindingVersion: typeof version;
}

/**
 * @see https://github.com/asyncapi/bindings/tree/master/websockets
 * @desc Protocol-specific definitions for an operation.
 * @desc This object MUST NOT contain any properties. Its name is reserved for future use.
 */
export interface SocketIOOperationBinding {}

/**
 * @desc Protocol-specific definitions for a message.
 * @desc This object MUST NOT contain any properties. Its name is reserved for future use.
 * */
export interface SocketIOMessageBinding {
  /** @desc Acknowledgement schema */
  ack?: SchemaObject | ReferenceObject;
  bindingVersion: typeof version;
}
