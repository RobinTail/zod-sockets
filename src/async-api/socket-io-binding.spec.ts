import { describe, expect, test } from "vitest";
import { version } from "./socket-io-binding";

describe("Socket.IO binding for AsyncAPI", () => {
  describe("Version", () => {
    test("Should have certain value", () => {
      expect(version).toMatchSnapshot();
    });
  });
});
