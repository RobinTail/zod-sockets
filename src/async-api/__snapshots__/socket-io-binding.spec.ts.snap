// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Socket.IO binding for AsyncAPI > Version > Should have certain value 1`] = `"0.11.0"`;
